thread/thread.cpp

```diff
@@ -934,13 +935,14 @@ R"(
         if (unlikely(!rq.current))
             LOG_ERROR_RETURN(ENOSYS, nullptr, "Photon not initialized in this vCPU (OS thread)");
         size_t randomizer = (rand() % 32) * (1024 + 8);
-        stack_size = align_up(randomizer + stack_size + sizeof(thread), PAGE_SIZE);
+        stack_size = align_up(randomizer + stack_size + sizeof(thread) + PAGE_SIZE, PAGE_SIZE);
         char* ptr = (char*)photon_thread_alloc(stack_size);
+        mprotect(ptr, PAGE_SIZE, PROT_NONE);
```

Comment:
is ```ptr``` guaranteed to be aligned to PAGE_SIZE?

Reply:
> is `ptr` guaranteed to be aligned to PAGE_SIZE?

That is a problem. I will add extra pages allocation and make  protect page aligned.